docs/library/esp32.rst

```diff
@@ -145,6 +145,98 @@ Constants
+
+.. method:: PCNT.init(*, ...)
+
+    Supported keyword arguments are:
+
+      - ``pin``: the input Pin to monitor for pulses
+      - ``rising``: an action to take on a rising edge - one of
+        ``PCNT.INCREMENT``, ``PCNT.DECREMENT`` or ``PCNT.IGNORE`` (the default)
+      - ``falling``: an action to take on a falling edge
+      - ``mode_pin``: ESP32 pulse counters support monitoring a second pin and
+        altering the behaviour of the counter based on its level - set this
+        keyword to any input Pin
+      - ``mode_low``: set to either ``PCNT.HOLD`` or ``PCNT.REVERSE`` to
+        either suspend counting or reverse the direction of the counter (i.e.,
+        ``PCNT.INCREMENT`` behaves as ``PCNT.DECREMENT`` and vice versa)
+        when ``mode_pin`` is low
+      - ``mode_high``: as ``mode_low`` but for the behaviour when ``mode_pin``
+        is high
+      - ``filter``: set to a value 1..1023, in ticks of the 80MHz clock, to
+        enable the pulse width filter
+      - ``minimum``: set to the minimum level of the counter value when
+        decrementing (-32768..0)
+      - ``maximum``: set to the maximum level of the counter value when
+        incrementing (0..32767)
+      - ``threshold0``: sets the counter value for the
+        ``PCNT.IRQ_THRESHOLD0`` event (see ``irq`` method)
+      - ``threshold1``: sets the counter value for the
+        ``PCNT.IRQ_THRESHOLD1`` event (see ``irq`` method)
+      - ``channel``: see description below
+
+    Each pulse counter unit supports two channels, 0 and 1. By default, the
+    constructor and ``init()`` method configure the 0 channel. Call the
+    ``init()`` method with ``channel=1`` to configure the other channel. Both
+    channels update the same counter value, but are able to monitor different
+    pins with different edge behaviour. The ``pin``, ``rising``, ``falling``,
+    ``mode_pin``, ``mode_low`` and ``mode_high`` keywords are per-channel, all
+    other keyword arguments are per-unit and specifying them will override any
+    previous configuration.
+
+    The second channel can be used to configure 4X quadrature decoding with a
+    single counter unit::
+
+        pin_a = Pin(2, Pin.INPUT, pull=Pin.PULL_UP)
+        pin_b = Pin(3, Pin.INPUT, pull=Pin.PULL_UP)
+        rotary = PCNT(0, pin=pin_a, falling=PCNT.INCREMENT, rising=PCNT.DECREMENT, mode_pin=pin_b, mode_low=PCNT.REVERSE)
+        rotary.init(channel=1, pin=pin_b, falling=PCNT.DECREMENT, rising=PCNT.INCREMENT, mode_pin=pin_a, mode_low=PCNT.REVERSE)
+        rotary.start()
+
+.. method:: PCNT.value([value])
+
+    Call this method with no arguments to return the current counter value or
+    pass the value 0 to reset the counter and return the value before reset.
+    ESP32 pulse counters do not support being set to any value other than 0.
+    Reset and return is not atomic and so it is possible for a pulse to be
+    missed.
+
+.. method:: PCNT.irq(trigger=Counter.IRQ_ZERO, handler=None)
```

Comment:
This should probably be PCNT.IRQ_ZERO

Reply:
Ta for catching these. I've found another and revised the `irq()` docs as well.